### PR TITLE
Use all `packages` locations when resetting canary changes

### PR DESCRIPTION
**Pull request by jwickens**
## Description
Iterate through custom packages locations instead of assuming `packages/*`

## Motivation and Context
My project has custom packages locations and all needs canary publish. See also #875 

## How Has This Been Tested?
Tests have been updated
## Types of changes
- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

### CODE REVIEW

**Comment by jwickens** [comment]
not sure why the checks are failing, getting a bunch of lint errors that i wasnt getting locally.

File: .gitignore [hunk 1]
```diff
@@ -23,7 +23,6 @@ composer.lock
 # Project Specific
 node_modules
 *.log
-lib
```
Comment (evocateur) [inline .gitignore]: Revert this, and remove all the `lib/*` files you've added in this PR. They are build output, and never under version control.

**Comment by jwickens** [comment]
Sorry about that, forgot how pull requests work :P !

File: src/commands/PublishCommand.js [hunk 2]
```diff
@@ -272,7 +272,9 @@ export default class PublishCommand extends Command {
       if (this.options.canary) {
         this.logger.info("canary", "Resetting git state");
         // reset since the package.json files are changed
-        GitUtilities.checkoutChanges("packages/*/package.json", this.execOpts);
+        this.repository.packageParentDirs.forEach(packageDir => {
+          GitUtilities.checkoutChanges(`${packageDir}/*/package.json`, this.execOpts);
```
Comment (evocateur) [inline src/commands/PublishCommand.js]: As enticing as the `packageParentDirs` are, we still need to support some weird configs (like `"."`, ugh) for package locations. We should iterate over `this.repository.packageConfigs`, joining the filename to the glob:

```js
this.repository.packageConfigs.forEach(pkgGlob => {
  GitUtilities.checkoutChanges(`${pkgGlob}/package.json`, this.execOpts);
});
```

The test will stay the same, and the existing tests are sufficient for this feature.

**Comment by jwickens** [comment]
ah ok that seems safer indeed

**Comment by evocateur** [comment]
Thanks!

**Pull request merged**